src/bun.js/api/bun/socket.zig

```diff
@@ -913,8 +913,8 @@ fn NewSocket(comptime ssl: bool) type {
                 err_value,
             });
 
-            _ = handlers.rejectPromise(err_value);
             if (!result.isEmptyOrUndefinedOrNull() and result.isAnyError(globalObject)) {
```

Comment:
isAnyError now checks `isEmptyOrUndefinedOrNull`
```suggestion
            if (result.isAnyError(globalObject)) {
```